### PR TITLE
NW6 | Fikret Ellek | React-Module-Project | Week-4 | Feature Validate AddBooking

### LINKED ISSUE #24: Feature Validate AddBooking
From React-Module-Project created by [SallyMcGrath](https://github.com/SallyMcGrath): CodeYourFuture/React-Module-Project#26  

Depends on #24 

**Instructions:** 

Add validation to some fields from #24 : the first name and last name must not be empty, the email must contain exactly 1 `@` symbol, and at least one `.` symbol after the `@`; the room ID must be a number between 0 and 100. If the fields do not contain correct information when the 'Submit' button is pressed, display a red error message at the top of the page, but do not clear the text already in the field.

**Test:** 

An invalid input displays an error message after the 'Submit' button is pressed (e.g. an email like `react@com` is invalid). A valid input shows the correct values at the bottom of the page.

**Reflection:** 

Validating user input is an important part of any application. Without checking the input, you might see unexpected errors when working with the data later. 

What do you think would happen if you were asked to remove a booking for room number '81', but the user had typed 'eightyOne' or 'EIGHTY ONE'?

### PR DESCRIPTION
**Pull request by fikretellek**
## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

### CODE REVIEW

File: src/components/AddBookingForm/AddBookingForm.jsx [hunk 1]
```diff
@@ -23,21 +38,46 @@ const AddBookingForm = ({ bookings, setBookings }) => {
...
+    } else if (!(formData.roomId > 0 && formData.roomId < 100)) {
+      alert("Enter a valid roon id");
+    } else if (!(formData.checkInDate < formData.checkOutDate)) {
+      alert("Enter valid dates");
+    } else {
+      return true;
+    }
+  };
+
   const handleSubmit = (e) => {
```
Comment (fhkahin) [inline src/components/AddBookingForm/AddBookingForm.jsx:64]: Brilliant calling validateBooking function  inside the handleSubmit function.

**Review by fhkahin — APPROVED** [review]

File: src/components/AddBookingForm/AddBookingForm.jsx [hunk 2]
```diff
@@ -23,21 +38,46 @@ const AddBookingForm = ({ bookings, setBookings }) => {
     setFormData((prevData) => ({ ...prevData, [name]: value }));
   };
 
+  const validateBooking = () => {
+    if (!isNaN(parseInt(formData.firstName))) {
+      alert("Enter a valid name");
+    } else if (!isNaN(parseInt(formData.surname))) {
+      alert("Enter a valid surname");
```
Comment (Ara225) [inline src/components/AddBookingForm/AddBookingForm.jsx:45]: This is an intresting approach to the problem. I think it mostly works, even for special letters such as Ö which is good as few validation solutions do. Of course it includes ! and other special chars

File: src/components/AddBookingForm/AddBookingForm.jsx [hunk 3]
```diff
@@ -23,21 +38,46 @@ const AddBookingForm = ({ bookings, setBookings }) => {
...
+    } else if (
+      !(
+        formData.email.includes("@") &&
+        formData.email.includes(".") &&
+        formData.email.indexOf("@") === formData.email.lastIndexOf("@") &&
+        formData.email.indexOf("@") < formData.email.lastIndexOf(".")
+      )
```
Comment (Ara225) [inline src/components/AddBookingForm/AddBookingForm.jsx:52]: This works but it's fairly hard to read I normally use a regex in these cases but that is a diffrent thing. A fairly simple regex such as formData.email.match(/^.+@.+\.$/)) should work

File: src/components/AddBookingForm/AddBookingForm.jsx [hunk 4]
```diff
@@ -23,21 +38,46 @@ const AddBookingForm = ({ bookings, setBookings }) => {
...
+    ) {
+      alert("Enter a valid email");
+    } else if (!(formData.roomId > 0 && formData.roomId < 100)) {
+      alert("Enter a valid roon id");
+    } else if (!(formData.checkInDate < formData.checkOutDate)) {
+      alert("Enter valid dates");
```
Comment (Ara225) [inline src/components/AddBookingForm/AddBookingForm.jsx:58]: Good work, a more specfic error message here would be helpful to the user though - doesn't say which date is invalid

**Review by Ara225 — COMMENTED** [review]
Looks good

File: src/components/AddBookingForm/AddBookingForm.jsx (hunk 2 above)
Comment (fikretellek) [reply src/components/AddBookingForm/AddBookingForm.jsx:45]: as you said regex would cover it all. to get a valid name should I use [a-zA-z]?

**Pull request merged**